connmgr.go

```diff
@@ -315,14 +343,22 @@ func (nn *cmNotifee) Connected(n inet.Network, c inet.Conn) {
 	cm.lk.Lock()
 	defer cm.lk.Unlock()
 
-	pinfo, ok := cm.peers[c.RemotePeer()]
+	id := c.RemotePeer()
+	pinfo, ok := cm.peers[id]
 	if !ok {
 		pinfo = &peerInfo{
+			id:        id,
 			firstSeen: time.Now(),
 			tags:      make(map[string]int),
 			conns:     make(map[inet.Conn]time.Time),
 		}
-		cm.peers[c.RemotePeer()] = pinfo
+		cm.peers[id] = pinfo
+	} else if pinfo.temp {
+		// we had created a temporary entry for this peer to buffer early tags before the
+		// Connected notification arrived: flip the temporary flag, and update the firstSeen
+		// timestamp to the real one.
+		pinfo.temp = false
+		pinfo.firstSeen = time.Now()
```

Comment:
Technically, I'm not sure if we should bump this. We clearly _saw_ the connection before.

(however, this isn't really an issue)

Reply:
My rationale is that once you allow early tagging, you don't know when the protocol will actually use that API. I could imagine the DHT tagging a peer it wants to query before it connects to it, in which case there could be a larger delay between the insertion of the early tag and the connection establishment, so I chose to be resilient to that but we can revisit if it becomes a problem.